File: WhichPharmaPortal/WhichPharmaPortal/ClientApp/src/models/ProcessingSettings.ts
import { fromUnixTime } from "date-fns/esm";
import { ProcessingPformsSettings } from "./ProcessingPformsSettings";
import { ProcessingCatSettings } from "./ProcessingCatSettings";
import { ProcessingATCSettings } from "./ProcessingATCSettings";

export interface ProcessingSettings {
    culture: string;
    decimalSeparator: string;
    numberGroupSeparator: string;
    administrationFormsMap: { [key: string]: string[]; };
    drugFormsMap: { [key: string]: string[]; };
    drugFormsMap2: { [key: string]: ProcessingPformsSettings; };
    dciMap: { [key: string]: ProcessingPformsSettings; };
    catsMap: { [key: string]: ProcessingCatSettings; };
    atcMap: { [key: string]: ProcessingATCSettings; };
    pharmaUnitsMap: { [key: string]: string[]; };
    administrationFormPresenceFields: string[];
    drugFormPresenceFields: string[];
    strengthPresenceFields: string[];
    packagePresenceFields: string[];
}

File: WhichPharmaPortal/WhichPharma.Models/Client/ProcessingSettings.cs
using System.Collections.Generic;

namespace WhichPharmaPortal.Models.Client
{
    public class ProcessingSettings
    {
        public string Culture { get; set; }
        public string DecimalSeparator { get; set; }
        public string NumberGroupSeparator { get; set; }
        public IDictionary<string, string[]> AdministrationFormsMap { get; set; }
        public IDictionary<string, string[]> DrugFormsMap { get; set; }
        public IDictionary<string, string[]> PharmaUnitsMap { get; set; }
        public string[] AdministrationFormPresenceFields { get; set; }
        public string[] DrugFormPresenceFields { get; set; }
        public string[] StrengthPresenceFields { get; set; }
        public string[] PackagePresenceFields { get; set; }
        public Dictionary<string, HarmonizedPform> DrugFormsMap2 { get; set; }
        public Dictionary<string, HarmonizeCatForm> CatsMap { get; set; }
        public Dictionary<string, HarmonizedPform> DciMap { get; set; }
        public Dictionary<string, HarmonizedATC> ATCMap { get; set; }
    }
}
